plan/task/exec.go

```diff
@@ -277,16 +278,43 @@ func (e execCommon) runOpts() ([]llb.RunOption, error) {
 		llb.Dir(e.workdir),
 		llb.User(e.user),
 	}
+
 	for k, v := range e.hosts {
 		opts = append(opts, llb.AddExtraHost(k, net.ParseIP(v)))
 	}
+
 	for _, mnt := range e.mounts {
 		opt, err := mnt.runOpt()
 		if err != nil {
 			return nil, err
 		}
 		opts = append(opts, opt)
 	}
+
+	// Handle HTTP_PROXY, HTTPS_PROXY, FTP_PROXY, etc...
+	proxyEnv := llb.ProxyEnv{}
+	for _, envVar := range os.Environ() {
+		split := strings.SplitN(envVar, "=", 2)
+		if len(split) != 2 {
+			continue
+		}
+		key, val := strings.ToLower(split[0]), split[1]
+		switch key {
+		case "no_proxy":
```

Comment:
[maybe use `EqualFold` here since it's what buildkit uses? ](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/dockerfile2llb/convert.go#L1436-L1465)

Reply:
good point, I converted the switch case to ifs to use EqualFold.